src/Form/PrintArticleTypeForm.php

```diff
@@ -89,4 +103,35 @@ public function save(array $form, FormStateInterface $form_state) {
     $form_state->setRedirectUrl($print_article_type->toUrl('collection'));
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  protected function actions(array $form, FormStateInterface $form_state) {
+
+    $actions = parent::actions($form, $form_state);
+
+    if ($actions['delete']) {
+      $actions['delete']['#access'] = ($this->getEntityCount()) ? FALSE : TRUE;
```

Comment:
With this approach, the template can still be deleted with using the dropdown on `admin/structure/print_article_type` or directly navigating to `http://tfp.dev/admin/structure/print_article_type/.../delete`.
![Screenshot of admin/structure/print_article_type](http://share.undpaul.de/bilder/jh/Snippet_Template__Thunder_2017-04-12_05-57-32--t6fs6.png)

Another approach would be to alter the DeleteForm.